parachain/primitives/core/src/lib.rs

```diff
@@ -31,6 +31,10 @@ where
 	SiblingParaId::from(para_id).into_account_truncating()
 }
 
+pub fn sibling_sovereign_account_raw(para_id: ParaId) -> [u8; 32] {
+	SiblingParaId::from(para_id).into_account_truncating()
+}
+
```

Comment:
Rather, if the compiler allows it, make the output type a generic parameter, so we only need one version of `sibling_sovereign_account`:

```suggestion
pub fn sibling_sovereign_account<T>(para_id: ParaId) -> T {
	SiblingParaId::from(para_id).into_account_truncating()
}
```

Then that allows generalized behavior:
```rust
let x: T::AccountId = sibling_sovereign_account(para_id);
let y: [u8; 32] = sibling_sovereign_account(para_id);
```

Reply:
This doesnt work. See comment: https://github.com/Snowfork/polkadot-sdk/pull/23#discussion_r1387877212

Reply:
Ah oops! Sorry I missed your detailed explanation in the other pr.

Reply:
Resolving then.